DistroLauncher/Ubuntu/InitTasks.cpp

```diff
@@ -0,0 +1,427 @@
+#include <stdafx.h>
+#include "InitTasks.h"
+
+#include <algorithm>
+#include <charconv>
+#include <exception>
+#include <filesystem>
+#include <fstream>
+#include <sstream>
+#include <utility>
+#include <vector>
+#include <system_error>
+
+namespace Ubuntu {
+
+namespace {
+// Blocks the current thread until all initialization tasks finish.
+void waitForInitTasks(WslApiLoader& api);
+
+// Enforces the existence of a default WSL user either:
+// - defined in /etc/wsl.conf (which might not be in effect yet)
+// - defined in WSL API/registry
+// - or the lowest UID >= 1000 in the NSS passwd database
+// Returns false if a default user couldn't be set.
+bool enforceDefaultUser(WslApiLoader& api);
```

Comment:
The enforcement of declaration order is a feature of C-derived languages :)

You'll see me moving declarations around exactly for that. We could just declare all the things on top (is one way), but I like to keep declarations closer to their usage as much as possible, otherwise they just become noise.